File: 0367-valid-perfect-square/0367-valid-perfect-square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        s=1
        e=num
        while(s<=e):
            m=(s+e)//2
            mid=m*m
            if(mid==num):
                return True
            elif(mid<num):
                s=m+1
            else:
                e=m-1
        return False

File: 0367-valid-perfect-square/0367-valid-perfect-square.java
class Solution {
    public boolean isPerfectSquare(int num) {
        long s=1;
        long e=num;
        while(s<=e)
        {
            long m=(s+e)/2;
            long mid=m*m;
            if(mid==num)
            {
                return true;
            }
            else if(mid<num)
            {
                s=m+1;
            }
            else
            {
                e=m-1;
            }
        }
        return false;
    }
}